### PR TITLE
Add a metric for the compaction interval config value.

### PR DESCRIPTION
**Pull request by cstyan**
Adds a metric to the compactor that exposes the configured `CompactionInterval` as a gauge, in seconds. The intention is to be able to use this metric, the last successful compaction timestamp metric, and `time()` to more accurately calculate the # of compactions in a row that have failed. This would improve [this](https://github.com/grafana/cortex-jsonnet/blob/main/cortex-mixin/alerts/compactor.libsonnet#L66-L80) alert in cortex-jsonnet which is hardcoded to look for > 1 failures in a 2h time period.

I added a quick check to the first test I found that actually started the compactor, just to confirm the metric was being set. If you'd like a more indepth test, or to move that check to it's own test, just let me know.

Signed-off-by: Callum Styan <callumstyan@gmail.com>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/compactor/compactor.go
Comment (56quarters) [inline pkg/compactor/compactor.go:306]: Would this be a good candidate for a [GaugeFunc](https://pkg.go.dev/github.com/prometheus/client_golang/prometheus#GaugeFunc) metric?

Comment (cstyan) [reply pkg/compactor/compactor.go:306]: Good question. I supposed it would be if we currently, or plan to, support live reloading of this config value (and others obviously) on the compactor. If not, would GaugeFunc get us anything over the way I'm doing it here?

Comment (pracucci) [reply pkg/compactor/compactor.go:306]: No big preference, but I would set the metric value here in `newCompactor()` and not at the end of `starting()`. If I'm not missing anything, metrics can be scraped while starting as well.

**Review by pracucci — APPROVED** [review]
LGTM (modulo a nit)

File: pkg/compactor/compactor.go
Comment (pracucci) [inline pkg/compactor/compactor.go:307]: Maybe `cortex_compactor_compaction_interval_seconds` to keep it consistent with the config option name `compaction_interval`?

File: pkg/compactor/compactor.go
Comment (56quarters) [reply pkg/compactor/compactor.go:306]: No preference here either, I don't think a GaugeFunc would buy you anything beyond the config reload thing you mentioned. My thinking was really just to remove the call to set the value of the gauge which seemed a bit far away from the place the metric was defined :+1:

**Review by pstibrany — APPROVED** [review]
Thank you.

**Review by pracucci — COMMENTED** [review]
@cstyan Before merging, could you add a CHANGELOG entry, please? Looks like an `[ENHANCEMENT]`.

**Comment by cstyan** [comment]
Changelog entry added

@pracucci @pstibrany @56quarters thanks for the reviews

**Pull request merged**
